spinalcordtoolbox/centerline/optic.py

```diff
@@ -72,22 +127,27 @@ def detect_centerline(image_fname, contrast_type,
     nib.save(img, centerline_optic_RPI_filename)
 
     # reorient the output image to initial orientation
-    centerline_optic_filename = sct.add_suffix(file_data + ext_data,
-                                               "_centerline_optic")
+    centerline_optic_filename = sct.add_suffix(file_data + ext_data, "_centerline_optic")
     cmd_reorient = 'sct_image -i "%s" -o "%s" -setorient "%s" -v 0' % \
                    (centerline_optic_RPI_filename,
                     centerline_optic_filename,
                     image_input_orientation)
     sct.run(cmd_reorient, verbose=0)
 
     # copy centerline to parent folder
+    folder_output_from_temp = folder_output
+    if not os.path.isabs(folder_output):
+        folder_output_from_temp = '../' + folder_output
+
     sct.printv('Copy output to ' + folder_output, verbose=0)
-    if os.path.isabs(folder_output):
-        shutil.copy(centerline_optic_filename, folder_output)
-    else:
-        dst_path = os.path.join("../", folder_output)
-        shutil.copy(centerline_optic_filename,
-                    '../' + folder_output)
+    shutil.copy(centerline_optic_filename, folder_output_from_temp)
+
+    if output_roi:
+        fname_roi_centerline = centerline2roi(fname_image=centerline_optic_RPI_filename,
+                                              folder_output=folder_output_from_temp,
+                                              verbose=verbose)
+
+        shutil.copy(reoriented_image_filename_nii, folder_output_from_temp)
```

Comment:
That is part of a discussion we had wiith @charleygros. The .roi centerline is defined (this is mandatory!) in RPI orientation, even if the original input image is not. So to be used in JIM, we need to have this RPI image along with the .roi centerline.

This can be subject to discussion but I would not spend too much time on it.